docs/docs/reference/augments/method-augments.md

```diff
@@ -0,0 +1,189 @@
+  def longestStrings: Seq[String] = {
+    val maxLength = this.map(_.length).max
+    this.filter(_.length == maxLength)
+  }
+}
+```
+
+### Augmented Type Patterns
+
+The previous example augmented a specific instance of a generic type. It is also possible
+to augment a generic type itself, using a _type pattern_:
+
+```scala
+augment List[type T] {
+  def second: T = this.tail.head
+}
+```
+
+The `type T` argument indicates that the augment applies to `List[T]`s for any type `T`.
+We also say that `type T` introduces `T` as a _variable_ in the type pattern `List[type T]`.
+Type variables may appear anywhere in a type pattern. Example:
+
+```scala
+augment List[List[type T]] {
+  def flattened: List[T] = this.foldLeft[List[T]](Nil)(_ ++ _)
+}
+```
+
+### Type Patterns in Cases
+
+The `type ...` syntax for pattern bound type variables also applies to patterns in
+case clauses of `match` and `try` expressions. For instance:
+
+```scala
+def f[T](s: Set[T], x: T) = s match {
+  case _: SortedSet[type U] => ... // binds `U`, infers that `U = T`
+  case _ =>
+}
+```
+
+Previously, one used a lower-case name to indicate a variable in a type pattern, as in:
+
+```scala
+  case _: SortedSet[u] => ... // binds `u`, infers that `u = T`
+```
+
+While being more regular wrt term variables in patterns, this usage is harder to read, and has the problem that it feels unnatrual to have to write type names in lower case. It will therefore be phased out to be replaced by the explicit `type T` syntax.
+
+Type patterns in cases only come in unbounded form; the bounds defined in the next section are not applicable to them.
+
+### Bounds in Augmented Type Patterns
+
+It is also possible to use bounds for the type variables in an augmented type pattern. Examples:
+
+```scala
+augment List[type T <: Shape] {
+  def totalArea = this.map(_.area).sum
+}
+```
+
+Context-bounds are also supported:
+
+```scala
+augment Seq[type T: math.Ordering] {
+  def indexOfLargest  = this.zipWithIndex.maxBy(_._1)._2
+  def indexOfSmallest = this.zipWithIndex.minBy(_._1)._2
+}
+```
+
+### Implicit Parameters for Type Patterns
+
+The standard translation of context bounds expands the bound in the last example to an implicit _evidence_ parameter of type `math.Ordering[T]`. It is also possible to give evidence parameters explicitly. The following example is equivalent to the previous one:
+
+```scala
+augment Seq[type T](implicit ev: math.Ordering[T]) {
+  def indexOfLargest  = this.zipWithIndex.maxBy(_._1)._2
+  def indexOfSmallest = this.zipWithIndex.minBy(_._1)._2
+}
+```
+
+There can be only one parameter clause following a type pattern and it must be implicit. As usual, one can combine context bounds and implicit evidence parameters.
+
+### Toplevel Type Variables
+
+A type pattern consisting of a top-level typevariable introduces a fully generic augmentation. For instance, the following augment introduces `x ~ y` as an alias
+for `(x, y)`:
+
+```scala
+augment (type T) {
```

Comment:
Why parens instead of brackets?

~~~ scala
augment [type T] {
  def ~ [U](that: U) = (this, that)
}
~~~

~~~ scala
augment [type F[type A]](implicit functor: Functor[F]) {
  def map[B](f: A => B): F[B] = functor.map(this, f)
}
~~~

Reply:
It's just precedence - the type pattern syntax requires a sub-production of types. So to get a top-level `type` you need to parents. By contrast `[...]` would be something new and magic.